src/qibolab/transpilers/transpile.py

```diff
@@ -66,19 +91,26 @@ def can_execute(circuit, verbose=True):
 
         if len(gate.qubits) == 1:
             if not isinstance(gate, (gates.I, gates.Z, gates.RZ, gates.U3)):
-                if verbose:
-                    vlog(f"{gate.name} is not native gate.")
+                vlog(f"{gate.name} is not a single qubit native gate.")
                 return False
 
         elif len(gate.qubits) == 2:
-            if not isinstance(gate, gates.CZ):
-                vlog(f"{gate.name} is not native gate.")
+            if CZ_is_native and iSWAP_is_native:
+                if not isinstance(gate, (gates.CZ, gates.iSWAP)):  # pragma: no cover
+                    vlog(f"{gate.name} is not a two qubit native gate.")
+                    return False
+            elif CZ_is_native:
+                if not isinstance(gate, (gates.CZ)):  # pragma: no cover
+                    vlog(f"{gate.name} is not a two qubit native gate.")
+                    return False
+            elif iSWAP_is_native:
+                if not isinstance(gate, (gates.iSWAP)):  # pragma: no cover
+                    vlog(f"{gate.name} is not a two qubit native gate.")
+                    return False
+            else:  # pragma: no cover
+                vlog("Use only CZ and/or iSWAP as native gates")
                 return False
-            if 0 not in gate.qubits:
-                vlog("Circuit does not respect connectivity. " f"{gate.name} acts on {gate.qubits}.")
-                return False
```

Comment:
Here you removed the check of connectivity. If a two-qubit gate does not contain the middle qubit then it is not native, even if it is CZ/iSWAP. 

Previously we were assuming that the middle qubit is 0, however this should be changed to 2 for our current chips. I was planning to open another PR for this change.

Reply:
i removed it by mistake, i will put it again, if you want to change everything to have middle qubit n.2 i can try to do it.

Reply:
You can give a try if you like. I think you only need to change all
```py
if 0 not in gate.qubits
```
to
```py
if 2 not in gate.qubits
```
in transpile.py and connectivity.py.

Reply:
It is not too easy, when middle qubit is 2 and circuit has less than 3 qubits problems in mapping occur as the hardware circuit must contain a number of qubits greater than that of the circuit to be transpiled...

Reply:
You are right, let's continue this discussion in #257.